### DISCUSSION TITLE
Dispatch event in Livewire Form

### DISCUSSION THREAD

**Discussion by mansoorkhan96** [livewire/livewire#8805] [1. Help] [answered]
I have a Livewire Form that i am using a component. I am try to find a way to dispatch events from the Form class.

```php
use Livewire\Form;

class ActivityForm extends Form
{
    public function boot()
    {
        $this->withValidator(function (Validator $validator) {
            if ($validator->fails()) {
                // Dispatch an event here, but we dont have event handler in here. 
                // $this->dispatch('activity-form-failed');
            }
        });
    }
}
```

**Comment by joshhanley** [comment, marked as answer]
@mansoorkhan96 try using `$this->component->dispatch()`.

**Reply by mansoorkhan96** [reply]
Thanks Josh `$this->component->dispatch()` works.

Is it documented? I could not find in docs.

**Reply by shankhadevpadam** [reply]
@joshhanley Is it possible to set the property in the component? I have a property in the component

```
public ?int $step = null;
```

Now I want to reset the value from the Form method

```
public function store()
{
     ///    
}
```
